packages/common/src/api/tan-query/wallets/useSendTokens.ts

```diff
@@ -0,0 +1,182 @@
+import { AudioWei } from '@audius/fixed-decimal'
+import { useMutation, useQueryClient } from '@tanstack/react-query'
+
+import { useQueryContext } from '~/api/tan-query/utils'
+import { Name, SolanaWalletAddress } from '~/models'
+import { getErrorMessage } from '~/utils'
+
+import { getUserCoinQueryKey } from '../coins/useUserCoin'
+import { useWalletAddresses } from '../users/account/useWalletAddresses'
+
+import { useTokenBalance } from './useTokenBalance'
+
+export type SendTokensParams = {
+  recipientWallet: SolanaWalletAddress
+  amount: AudioWei
+}
+
+export type SendTokensResult = {
+  signature: string
+  success: boolean
+}
+
+/**
+ * Hook for sending tokens on Solana blockchain.
+ * This hook handles only Solana transfers, not ETH transfers.
+ *
+ * @returns Mutation object with sendTokens function and status
+ */
+export const useSendTokens = ({ mint }: { mint: string }) => {
+  const queryClient = useQueryClient()
+  const { audiusBackend, audiusSdk, reportToSentry, analytics, env } =
+    useQueryContext()
+  const { data: walletAddresses } = useWalletAddresses()
+
+  // Get current user's token balance using the existing hook
+  const { data: tokenBalance } = useTokenBalance({ mint })
+
+  return useMutation({
+    mutationFn: async ({
+      recipientWallet,
+      amount
+    }: SendTokensParams): Promise<SendTokensResult> => {
+      try {
+        // For now, we only support wAUDIO transfers
+        // This can be extended to support other tokens in the future
+        if (mint !== env.WAUDIO_MINT_ADDRESS) {
+          throw new Error(`Token mint ${mint} is not supported for sending`)
+        }
+
+        // Get current user's ETH address
+        const currentUser = walletAddresses?.currentUser
+        if (!currentUser) {
+          throw new Error('Failed to retrieve current user wallet address')
+        }
+
+        // Get SDK instance
+        const sdk = await audiusSdk()
+
+        // Check if user has sufficient balance using the hook data
+        // tokenBalance.balance is a FixedDecimal, so we need to access its .value property
+        if (!tokenBalance?.balance || tokenBalance.balance.value < amount) {
+          throw new Error('Insufficient balance to send tokens')
+        }
+
+        // Send the tokens using the audius backend
+        // This method throws errors instead of returning them
+        await audiusBackend.sendWAudioTokens({
```

Comment:
so we don't have a send in sdk yet?

Reply:
i think it technically uses sdk but is a weird wrapper function